### PR TITLE
Prepare release 2.3.7

### PR DESCRIPTION
**Pull request by mcmtroffaes**
@graingert Anything else needed that you can think of? Can you think of any good upstream projects or high profile users which need this package that might be interested to help testing this?

### CODE REVIEW

File: CHANGELOG.rst
Comment (graingert) [inline CHANGELOG.rst:13]: There's no py.typed file so the annotations won't be seen

Comment (mcmtroffaes) [reply CHANGELOG.rst:13]: That's intentional. Those annotations have been added for internal use only, mainly to verify unicode behaviour between functions. The "official" type annotations live in typeshed.

**Comment by mcmtroffaes** [comment]
Is this good to merge (conditional on #69 being merged)? I'm still waiting for a response on your ``py.typed`` related comment @graingert

**Comment by mcmtroffaes** [comment]
A heads up that I'll be merging this shortly (unless someone still wants to raise an issue).

File: CHANGELOG.rst
Comment (graingert) [reply CHANGELOG.rst:13]: Oh ok I think it's worth moving the types in here soon then

**Review by graingert — APPROVED** [review]

**Comment by mcmtroffaes** [comment]
Will be tagging in a moment, we'll find out if the release script does it's job as intended... 🤞

File: CHANGELOG.rst
Comment (mcmtroffaes) [reply CHANGELOG.rst:13]: Yup, thanks!

**Comment by mcmtroffaes** [comment]
The release script failed:

https://github.com/jazzband/pathlib2/runs/5109176234

Not sure why. Anyway, I've done a quick manual release so it's on pypi.

**Comment by mcmtroffaes** [comment]
Just to note I've had to yank the release (and, the post release that tried to fix the problem), due to a dependency issue and python2's pip being too old to properly resolve dependencies (who knew...). I think for future pure python 3 releases we best bump the major version number for upstream projects still on python 2. I have some cleanup PR coming up which will take care of this.

**Pull request merged**
